src/algebra/module/pi.lean

```diff
@@ -115,6 +115,11 @@ lemma single_smul {α} [monoid α] [Π i, add_monoid $ f i]
   single i (r • x) = r • single i x :=
 single_op (λ i : I, ((•) r : f i → f i)) (λ j, smul_zero _) _ _
 
+lemma single_smul'' {α β} [monoid α] [add_monoid β]
+  [distrib_mul_action α β] [decidable_eq I] (i : I) (r : α) (x : β) :
+  single i (r • x) = r • single i x :=
+single_smul i r x
```

Comment:
This could do with a comment explaining why it is needed when we already have `single_smul`